src/mlpack/tests/main_tests/linear_svm_test.cpp

```diff
@@ -0,0 +1,866 @@
+      CLI::GetParam<LinearSVM<>*>("output_model")->Parameters();
+
+  // Both solutions should be not equal.
+  CheckMatricesNotEqual(parameters1, parameters2);
+}
+
+/**
+  * Ensuring that delta has some effects on the output.
+ **/
+BOOST_AUTO_TEST_CASE(LinearSVMDiffDeltaTest)
+{
+  arma::mat trainData;
+  if (!data::Load("iris.csv", trainData))
+    BOOST_FAIL("Cannot load test dataset iris.csv!");
+
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  SetInputParam("training", trainData);
+  SetInputParam("labels", trainLabels);
+  SetInputParam("delta", double(1.0));
+
+  // First solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after first training.
+  const arma::mat parameters1 =
+      std::move(CLI::GetParam<LinearSVM<>*>("output_model")
+                ->Parameters());
+
+  // Reset the settings.
+  bindings::tests::CleanMemory();
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("delta", double(1000));
+
+  // Second solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after second training.
+  const arma::mat& parameters2 =
+      CLI::GetParam<LinearSVM<>*>("output_model")->Parameters();
+
+  // Both solutions should be not equal.
+  CheckMatricesNotEqual(parameters1, parameters2);
+}
+
+/**
+  * Ensuring that no_intercept has some effects on the output.
+ **/
+BOOST_AUTO_TEST_CASE(LinearSVMDiffInterceptTest)
+{
+  arma::mat trainData;
+  if (!data::Load("iris.csv", trainData))
+    BOOST_FAIL("Cannot load test dataset iris.csv!");
+
+  arma::Row<size_t> trainLabels;
+  if (!data::Load("iris_labels.txt", trainLabels))
+    BOOST_FAIL("Cannot load test dataset iris_labels.txt!");
+
+  SetInputParam("training", trainData);
+  SetInputParam("labels", trainLabels);
+
+  // First solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after first training.
+  const arma::mat parameters1 =
+      std::move(CLI::GetParam<LinearSVM<>*>("output_model")
+                ->Parameters());
+
+  // Reset the settings.
+  bindings::tests::CleanMemory();
+  CLI::ClearSettings();
+  CLI::RestoreSettings(testName);
+
+  SetInputParam("training", std::move(trainData));
+  SetInputParam("labels", std::move(trainLabels));
+  SetInputParam("no_intercept", bool(true));
+
+  // Second solution.
+  mlpackMain();
+
+  // Get the parameters of the output model obtained after second training.
+  const arma::mat& parameters2 =
+      CLI::GetParam<LinearSVM<>*>("output_model")->Parameters();
+
+  // Both solutions should be not equal.
+  CheckMatricesNotEqual(parameters1, parameters2);
+}
+
+/**
+ * The test is only compiled if the user has not specified OpenMP to be
+ * used and we test these case in one core.
+ */
+#ifndef HAS_OPENMP
+/**
```

Comment:
It shouldn't be necessary to check for OpenMP here---the test should work both with and without OpenMP.

Reply:
Removed this. Without OPEN_MP how can we ensure that it must run on one core because this OPEN_MP it will go `omp_get_max_threads()`.

Reply:
Hi @rcurtin I made the changes as suggested.